### PR TITLE
aggregate 'fail to resolve dependency reference' messages and change log level considering user has no action for it

### PR DESCRIPTION
**Pull request by ansyral**

### CODE REVIEW

**Comment by hellosnow** [comment]
:shipit:

File: src/Microsoft.DocAsCode.Build.Engine/Incrementals/DependencyGraph.cs [hunk 1]
```diff
@@ -385,6 +403,11 @@ private void ResolveReferenceCoreNoLock()
                     CreateOrUpdate(_indexOnTo, updated.To.Value, updated);
                 }
             }
+
+            if (unresolved.Count > 0)
+            {
+                Logger.LogVerbose($"Dependency graph Failed to resolve references: {JsonUtility.Serialize(unresolved)}.");
```
Comment (vicancy) [inline src/Microsoft.DocAsCode.Build.Engine/Incrementals/DependencyGraph.cs]: >JsonUtility.Serialize [](start = 84, length = 21)

use string.join?

Comment (vicancy) [inline src/Microsoft.DocAsCode.Build.Engine/Incrementals/DependencyGraph.cs]: >F [](start = 53, length = 1)

f

File: src/Microsoft.DocAsCode.Build.Engine/Incrementals/DependencyGraph.cs [hunk 2]
```diff
@@ -385,6 +403,11 @@ private void ResolveReferenceCoreNoLock()
                     CreateOrUpdate(_indexOnTo, updated.To.Value, updated);
                 }
             }
+
+            if (unresolved.Count > 0)
+            {
+                Logger.LogVerbose($"Dependency graph failed to resolve references: {string.Join(";", unresolved)}.");
```
Comment (vicancy) [inline src/Microsoft.DocAsCode.Build.Engine/Incrementals/DependencyGraph.cs]: resolve {unresolved.Count} references?

File: src/Microsoft.DocAsCode.Build.Engine/Incrementals/DependencyGraph.cs [hunk 3]
```diff
@@ -385,6 +403,11 @@ private void ResolveReferenceCoreNoLock()
                     CreateOrUpdate(_indexOnTo, updated.To.Value, updated);
                 }
             }
+
+            if (unresolved.Count > 0)
+            {
+                Logger.LogVerbose($"Dependency graph failed to resolve {unresolved.Count} references: {string.Join(";", unresolved)}.");
```
Comment (vwxyzh) [inline src/Microsoft.DocAsCode.Build.Engine/Incrementals/DependencyGraph.cs]: avoid long text in one line.

**Comment by superyyrrzz** [comment]
:shipit:

**Pull request merged**